core/src/test/scala/io/finch/EndToEndSpec.scala

```diff
@@ -53,4 +55,15 @@ class EndToEndSpec extends FinchSpec {
     rep.contentString shouldBe "bar"
     rep.status shouldBe Status.Created
   }
+
+  it should "convert Endpoint[AsyncStream[Buf]] to service" in {
```

Comment:
So when you get a response out, it has a `reader: Reader` method that allows you to read streaming chunks using the reader API. Instead of doing that, I suggest converting reader into `AsyncStream` via `AsynStream.fromReader(request.reader)` and then comparing two async stream (you can convert those to `toSeq()`).